### PR TITLE
#2407 - The ordering of branches of cascade reactions and reactions themselves should be the same on Load/Save from/to RDF

### PR DESCRIPTION
**Pull request by even1024**
## Generic request
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name does not contain '#'
- [x] base branch (master or release/xx) is correct
- [x] PR is linked with the issue
- [x] task status changed to "Code review"
- [x] code follows product standards
- [x] regression tests updated

### CODE REVIEW

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp [hunk 1]
```diff
@@ -236,15 +237,54 @@ void ReactionMultistepDetector::createSummBlocks()
...
+
+    // Sort the blocks based on the sorted indices
+    std::vector<MolSumm> sorted_blocks;
+    sorted_blocks.reserve(_component_summ_blocks.size());
+    for (auto idx : indices)
+        sorted_blocks.emplace_back(std::move(_component_summ_blocks[idx]));
+    _component_summ_blocks = std::move(sorted_blocks);
+
+    // Update arrows_to and arrows_from with new indices
+    for (auto& block : _component_summ_blocks)
```
Comment (AlexanderSavelyev) [inline core/indigo-core/reaction/src/reaction_multistep_detector.cpp:263]: seems to be error. Can't update loop element

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp [hunk 2]
```diff
@@ -302,12 +342,14 @@ bool ReactionMultistepDetector::detectArrows()
             // idx_cs_min_reac <-> idx_cs_min_prod
             csb_min_reac.arrows_to.push_back(idx_cs_min_prod);
             csb_min_prod.arrows_from.push_back(idx_cs_min_reac);
+            csb_min_reac.reaction_idx = i;
```
Comment (AlexanderSavelyev) [inline core/indigo-core/reaction/src/reaction_multistep_detector.cpp:345]: rename i (to component_idx) ?

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp [hunk 3]
```diff
@@ -236,15 +237,54 @@ void ReactionMultistepDetector::createSummBlocks()
...
+    for (auto idx : indices)
+        sorted_blocks.emplace_back(std::move(_component_summ_blocks[idx]));
+    _component_summ_blocks = std::move(sorted_blocks);
+
+    // Update arrows_to and arrows_from with new indices
+    for (auto& block : _component_summ_blocks)
+    {
+        for (auto& arrow : block.arrows_to)
+        {
+            if (arrow >= 0 && arrow < static_cast<int>(old_to_new.size()))
```
Comment (AlexanderSavelyev) [inline core/indigo-core/reaction/src/reaction_multistep_detector.cpp:267]: move static_cast<int>(old_to_new.size() outside the loop

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp [hunk 4]
```diff
@@ -359,18 +401,13 @@ bool ReactionMultistepDetector::detectMultitailArrows()
...
-                bad_pathway = true;
-        }
-
         // TODO: add upper limit
         if (idx_cs_min_prod < 0)
             bad_pathway = true;
         else
         {
             auto& csb_min_prod = _component_summ_blocks[idx_cs_min_prod];
+            csb_min_prod.reaction_idx = i;
```
Comment (AlexanderSavelyev) [inline core/indigo-core/reaction/src/reaction_multistep_detector.cpp:410]: rename i (to component_idx?)

**Review by AlexanderSavelyev — CHANGES_REQUESTED** [review]

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp (hunk 1 above)
Comment (even1024) [reply core/indigo-core/reaction/src/reaction_multistep_detector.cpp:263]: updated

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp (hunk 3 above)
Comment (even1024) [reply core/indigo-core/reaction/src/reaction_multistep_detector.cpp]: there is no need for this. it works as it should and static cast here is absolutely normal. static<cast> is not a function - this is an instruction to compiler on how to treat the data.

File: core/indigo-core/reaction/src/reaction_multistep_detector.cpp (hunk 4 above)
Comment (even1024) [reply core/indigo-core/reaction/src/reaction_multistep_detector.cpp]: fixed

**Review by AlexanderSavelyev — APPROVED** [review]

**Pull request merged**